pkg/wasm/imagefetcher.go

```diff
@@ -40,15 +43,20 @@ import (
 var errWasmOCIImageDigestMismatch = errors.New("fetched image's digest does not match the expected one")
 
 type ImageFetcherOption struct {
-	Username string
-	Password string
 	// TODO(mathetake) Add signature verification stuff.
-
-	Insecure bool
+	PullSecret []byte
+	Insecure   bool
```

Comment:
Is there a conflict with PullSecret  and insecure?

Reply:
Not really, insecure decides whether to verified server cert at client, pull secret is used at server side auth.